write-fonts/src/tables/layout.rs

```diff
@@ -36,9 +36,22 @@ macro_rules! lookup_type {
 /// lookups) can be given different lookup ids for each of GSUB/GPOS.
 macro_rules! table_newtype {
     ($name:ident, $inner:ident, $read_type:path) => {
+        /// A typed wrapper around a shared table.
+        ///
+        /// This is used so that we can associate the correct lookup ids for
+        /// lookups that are shared between GPOS/GSUB.
+        ///
+        /// You can access the inner type via `Deref` or the `as_inner` method.
```

Comment:
I worry that will be more confusing, since it's sort of an unusual pattern; I very rarely use `as _`, and would if for some reason I can't deref with `*`, (or if that is ambiguous) I would prefer to just call `.deref()` on the type directly.

In the case that the user is unfamiliar with `Deref`, the new `as_inner` method will appear in autocomplete, and that is an explicit way to achieve the intended effect, without needing any knowledge of `Deref` at all.

Reply:
> for reasons I do not understand

I feel mildly better that I'm not alone in finding how that behaved confusing

Reply:
amusing I think I tried &* but not &** because I thought the compiler expanded * to however many derefs it takes to work